### PR TITLE
fix(zvt): omit StatusEnquiry password when service byte is absent

### PR DESCRIPTION
**Pull request by 0xmuon**
## Summary

fixes `StatusEnquiry` serialization for ZVT 2.55 so the password BMP is only sent when a service byte is present,add `for_serialization()` to drop the password when service_byte is `None`,update call sites in `zvt_cli` and `zvt_feig_terminal`, derive Clone on both `StatusEnquiry` types, and add a unit test for password omitted vs included.

This fixes a real bug in production code: `Feig::status_enquiry()` was sending a password without a service byte.

## Tests

- [x] `cargo test -p zvt test_status_enquiry_for_serialization`
- [x] `cargo test --all`
- [ ] Manual: run `zvt_cli status --type zvt` against a terminal with and without `--service-byte`
- [ ] Manual: verify `Feig::status_enquiry()` still returns expected terminal status in production config -> for this production Config is needed, @ maintainers please verify this test.

### CODE REVIEW

**Comment by 0xmuon** [comment]
@dorezyuk could you please review the PRs.

**Comment by dorezyuk** [comment]
@0xmuon  yes on my todo list - sorry for the wait

File: zvt/src/packets.rs
Comment (dorezyuk) [inline zvt/src/packets.rs]: i think the fix should be moving this to a dedicated struct 
```rs
#[derive(Debug, Default, PartialEq, Zvt)]
pub struct StatusEnquiryInner {
    #[zvt_bmp(length = length::Fixed<3>, encoding = encoding::Bcd)]
    pub password: usize,

    #[zvt_bmp(number = 0x03)]
    pub service_byte: u8,
}

/// Chapter 2.55
#[derive(Debug, PartialEq, Zvt)]
#[zvt_control_field(class = 0x05, instr = 0x01)]
pub struct StatusEnquiry {
    pub inner: Option<StatusEnquiryInner>,

    #[zvt_bmp(number = 0x06, length = length::Tlv)]
    pub tlv: Option<tlv::StatusEnquiry>,
}
```

or similar - when doing so lets add a unit test the inner is None and Some. I've done it quickly and the test should looks somewhat like 
```rs
    #[rstest::rstest]
    fn test_status_enquiry() {
        let bytes = [5, 1, 0];
        let actual = StatusEnquiry::zvt_deserialize(&bytes).unwrap().0;
        let expected = StatusEnquiry {
            inner: None,
            tlv: None,
        };

        assert_eq!(actual, expected);
        assert_eq!(expected.zvt_serialize(), bytes);

        let bytes = [5, 1, 5, 0, 18, 52, 3, 2];
        let actual = StatusEnquiry::zvt_deserialize(&bytes).unwrap().0;
        let expected = StatusEnquiry {
            inner: Some(StatusEnquiryInner {
                password: 1234,
                service_byte: 2,
            }),
            tlv: None,
        };

        assert_eq!(actual, expected);
        assert_eq!(expected.zvt_serialize(), bytes);
    }
```

File: zvt/src/packets/tlv.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ pub struct StatusInformation {
     pub subs_on_card: Option<SubsOnCard>,
 }
 
-#[derive(Debug, Default, PartialEq, Zvt)]
+#[derive(Debug, Clone, Default, PartialEq, Zvt)]
 pub struct StatusEnquiry {
```
Comment (dorezyuk) [inline zvt/src/packets/tlv.rs:55]: i think we don't need the clone anymore

**Comment by dorezyuk** [comment]
looks good - just one small nit. can you fix and rebase?

**Comment by 0xmuon** [comment]
Sure!

**Review by dorezyuk — APPROVED** [review]
